### DISCUSSION TITLE
Request for Source code for Windows OS to build NanoMQ Gateway Tool

### DISCUSSION THREAD

**Discussion by anandbwfrd** [nanomq/nanomq#1182] [Q&A]
How can I subscribe to a topic from an external MQTT Broker and publish the content to a new topic on local MQTT Broker using the nanomq_cli tool on Windows OS?  I think it requires the usage of the NanoMQGateway tool which isn't built by default.  I need access to the Windows Source code to be able to build it using CMAKE and Ninja - Screenshot attached below.  Please confirm if the NanoMQGateway tool is supported on the Windows Platform.
![Screenshot 2023-04-18 201451](https://user-images.githubusercontent.com/112025627/232942152-be2bc2c0-b72c-4a06-87d5-a3de4474ec3e.png)

**Comment by JaylinYu** [comment]
It is mqtt bridging you are looking for.
Glad you figured it out by yourself.

**Comment by anandbwfrd** [comment]
Thanks a lot Jaylin for your help!!!!

Yes, tried out the MQTT Bridge option and it worked right away without any issues.

From: Jaylin ***@***.***>
Sent: Thursday, April 20, 2023 3:01 AM
To: emqx/nanomq ***@***.***>
Cc: Balasubramanian, Anand ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [emqx/nanomq] Request for Source code for Windows OS to build NanoMQ Gateway Tool (Discussion #1182)


CAUTION: This email originated from outside the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.

________________________________

It is mqtt bridging you are looking for.
Glad you figured it out by yourself.

-
Reply to this email directly, view it on GitHub<https://github.com/emqx/nanomq/discussions/1182#discussioncomment-5670489>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A2WWAG3HYUWYHWYEKEHGDY3XCDULHANCNFSM6AAAAAAXDLY6ZA>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

This electronic communication is for the use of the intended recipient(s) only, and may contain confidential, privileged or proprietary information. If you are not an intended recipient, please reply to the sender and then immediately delete and destroy all copies of the communication. See our Electronic Communications Terms and Conditions<https://www.weatherford.com/en/about-us/policies/terms-and-conditions-for-electronic-communications/> for further information
